lib/application_insights/rack/track_request.rb

```diff
@@ -82,6 +95,33 @@ def format_request_duration(duration_seconds)
 
         Time.at(duration_seconds).gmtime.strftime("00.%H:%M:%S.%7N")
       end
+
+      def request_id_header(env)
+        request_id_header = env['HTTP_REQUEST_ID']
+        valid_request_id_header = request_id_header && request_id_header[0] == '|'
+
+        length = valid_request_id_header ? 8 : 16
+        id = SecureRandom.hex(length / 2)
```

Comment:
Any reason using hex instead of base64 as documented here?
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/HierarchicalRequestId.md#root-request-id-generation

Reply:
I'm currently updating my implementation to include logs and also a change to Base64 :+1: